aws/resource_aws_elasticache_replication_group.go

```diff
@@ -861,23 +861,11 @@ func deleteElasticacheReplicationGroup(replicationGroupID string, timeout time.D
 	return err
 }
 
-func flattenElasticacheNodeGroupsToClusterMode(clusterEnabled bool, nodeGroups []*elasticache.NodeGroup) []map[string]interface{} {
-	if !clusterEnabled {
-		return []map[string]interface{}{}
-	}
-
-	m := map[string]interface{}{
-		"num_node_groups":         0,
-		"replicas_per_node_group": 0,
-	}
-
-	if len(nodeGroups) == 0 {
```

Comment:
This check is intentional to prevent a potential index reference panic where `nodeGroups` does not contain any elements.